apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotificationForOrganizationContextTask.js

```diff
@@ -0,0 +1,246 @@
+        to: { user: { id: resident.user } },
+        type: messageType,
+        meta: { dv: 1, data },
+        sender: { dv: 1, fingerprint: 'send-billing-receipts-added-notifications' },
+        uniqKey: notificationKey,
+        organization: { id: resident.organization },
+    }
+
+    const { isDuplicateMessage } = await sendMessage(context, messageData)
+
+    return isDuplicateMessage ? 0 : 1
+}
+
+/**
+ * Prepares data for sendMessage to resident on available billing receipt, then tries to send the message
+ * @param {BillingIntegrationOrganizationContext} context
+ * @param {string} lastSyncDate
+ * @returns {Promise<void>}
+ */
+async function sendBillingReceiptsAddedNotificationForOrganizationContext (context, lastSyncDate) {
+    const contextId = get(context, 'id')
+    if (!contextId) throw new Error(`Invalid BillingIntegrationOrganizationContext, cannot get context.id. Context: ${context}`)
+
+    const { keystone } = getSchemaCtx('Message')
+    const redisClient = getRedisClient()
+    const { thisMonthStart } = getStartDates()
+    const receiptCreatedAfter = lastSyncDate || thisMonthStart
+
+    const receipts = await fetchReceipts(contextId, receiptCreatedAfter)
+    logger.info({ msg: `Found ${receipts.length} receipts`, contextId })
+
+    const receiptAccountData = await prepareReceiptsData(receipts, context)
+    const serviceConsumers = await fetchServiceConsumers(context, receiptAccountData.accountNumbers)
+    const consumersByAccountKey = groupConsumersByAccountKey(serviceConsumers)
+
+    let successCount = 0
+    logger.info({ msg: 'Left receipts', receiptAccountData })
+
+    for (const [key, receipt] of Object.entries(receiptAccountData.receiptAccountKeys)) {
+        const consumers = consumersByAccountKey[key]
+        if (isEmpty(consumers)) continue
+
+        successCount += await notifyConsumers(redisClient, keystone, receipt, consumers)
+    }
+
+    logger.info({ msg: 'Sent billing receipts', successCount, contextId })
+}
+
+async function fetchReceipts (contextId, receiptCreatedAfter) {
+    const receiptsWhere = {
+        createdAt_gt: receiptCreatedAfter,
+        context: { id: contextId },
+        deletedAt: null,
+    }
+    return find('BillingReceipt', receiptsWhere)
+}
+
+async function prepareReceiptsData (receipts, context) {
+    const accountsNumbers = []
+    const receiptAccountKeys = {}
+
+    for (const receipt of receipts) {
+        if (await hasNewerReceipts(receipt)) continue
+
+        const processedReceipt = await processReceiptData(receipt, context)
+        const accountNumber = get(processedReceipt, 'account.number')
+        const addressKey = get(processedReceipt, 'account.property.addressKey')
+
+        accountsNumbers.push(accountNumber)
+        receiptAccountKeys[makeAccountKey(addressKey, accountNumber)] ||= processedReceipt
+    }
+
+    return { accountNumbers: accountsNumbers.filter(Boolean), receiptAccountKeys }
+}
+
+async function hasNewerReceipts (receipt) {
```

Comment:
Is it redundant check? BillingContext.lastReport changes only if newer receipts were loaded